jabgui/src/main/java/org/jabref/gui/externalfiles/UnlinkedFilesCrawler.java

```diff
@@ -91,7 +92,8 @@ FileNodeViewModel searchDirectory(Path directory, UnlinkedPDFFileFilter unlinked
         //   2. GitIgnoreFilter
         ChainedFilters filters = new ChainedFilters(unlinkedPDFFileFilter, new GitIgnoreFileFilter(directory));
         Map<Boolean, List<Path>> directoryAndFilePartition;
-        try (Stream<Path> filesStream = StreamSupport.stream(Files.newDirectoryStream(directory, filters).spliterator(), false)) {
```

Comment:
Why is it split in two ?

Reply:
I believe, autoclosable resource is the DirectoryStream. This would make sense to make it better readable and maintainable.